lib/models/instrumentation.js

```diff
@@ -0,0 +1,277 @@
+'use strict';
+
+var fs = require('fs-extra');
+var chalk = require('chalk');
+var experiments = require('../experiments/');
+var tree = require('heimdalljs-tree');
+var utilsInstrumentation = require('../utilities/instrumentation');
+
+var vizEnabled = utilsInstrumentation.vizEnabled;
+var instrumentationEnabled = utilsInstrumentation.instrumentationEnabled;
+
+function _enableFSMonitorIfInstrumentationEnabled() {
+  var monitor;
+  if (instrumentationEnabled()) {
+    var FSMonitor = require('heimdalljs-fs-monitor');
+    monitor = new FSMonitor();
+    monitor.start();
+  }
+  return monitor;
+}
+
+_enableFSMonitorIfInstrumentationEnabled();
+
+
+function Instrumentation (options) {
+  this.ui = options.ui;
+
+  // project constructor will set up bidirectional link
+  this.project = null;
+
+  this.instrumentations = {
+    init: options.initInstrumentation,
+    build: {
+      cookie: null,
```

Comment:
Should we call this something else? I feel like the cookie verbiage is removed from heimdall 0.3, and I'd prefer to avoid it here...

Maybe "token"?  What is it called in the 0.3 API?